File: android/src/main/kotlin/com/shareinvest/flutter_kakao_maps/model/ScaleBarOptions.kt
package com.shareinvest.flutter_kakao_maps.model

import org.json.JSONObject

data class ScaleBarOptions(
    var enabled: Boolean,
    var position: KakaoMapPosition,
    var autoDisabled: Boolean,
    var fadeInOutOptions: FadeInOutOptions,
)


fun JSONObject.toScaleBarOptions(): ScaleBarOptions {
    return ScaleBarOptions(
        this.getBoolean("enabled"),
        this.getJSONObject("position").toKakaoMapPosition(),
        this.getBoolean("autoDisabled"),
        this.getJSONObject("fadeInOutOptions").toFadeInOutOptions(),
    )
}

File: ios/flutter_kakao_maps/Sources/flutter_kakao_maps/model/ScaleBarOptions.swift
import KakaoMapsSDK

struct ScaleBarOptions {
    var enabled: Bool
    var position: KakaoMapPosition
    var autoDisabled: Bool
    var fadeInOutOptions: FadeInOutOptions
}

extension NSDictionary {
    func toScaleBarOptions() -> ScaleBarOptions {
        return ScaleBarOptions(
            enabled: self["enabled"] as! Bool,
            position: (self["position"] as! NSDictionary).toKakaoMapPosition(),
            autoDisabled: self["autoDisabled"] as! Bool,
            fadeInOutOptions: (self["fadeInOutOptions"] as! NSDictionary).toFadeInOutOptions()
        )
    }
}
